sql/core/src/test/resources/sql-tests/inputs/json-functions.sql

```diff
@@ -58,5 +58,16 @@ select schema_of_json('{"c1":01, "c2":0.1}', map('allowNumericLeadingZeros', 'tr
 select schema_of_json(null);
 CREATE TEMPORARY VIEW jsonTable(jsonField, a) AS SELECT * FROM VALUES ('{"a": 1, "b": 2}', 'a');
 SELECT schema_of_json(jsonField) FROM jsonTable;
+
+-- json_array_length
+select json_array_length('');
+select json_array_length('[]');
+select json_array_length('[1,2,3]');
+select json_array_length('[[1,2],[5,6,7]]');
+select json_array_length('[{"a":123},{"b":"hello"}]');
+select json_array_length('[1,2,3,[33,44],{"key":[2,3,4]}]');
+select json_array_length('{"key":"not a json array"}');
```

Comment:
Does this function need to be array-specific? Yea, I know this is the same behaviour with pgSQL. But, `json_length` in mysql is more general one? 
```
mysql> select json_length('[1,2,3]');
+------------------------+
| json_length('[1,2,3]') |
+------------------------+
|                      3 |
+------------------------+
1 row in set (0.01 sec)

mysql> select json_length('{"key":"not a json array"}');
+-------------------------------------------+
| json_length('{"key":"not a json array"}') |
+-------------------------------------------+
|                                         1 |
+-------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
I agree on the fact that `json_length` is generic. I defined it array-specific because most of the DBs support array-specific function only.

Reply:
WDYT? @HyukjinKwon

Reply:
The goal of adding of the `json_array_length` function is to make migration from other DBMS to Spark SQL easier. I think we should focus on the specific function `json_array_length` already provided by others.

Reply:
@iRakson . Could you fix the UT failure on this file?